react/src/pages/EndpointListPage.tsx

```diff
@@ -528,30 +538,36 @@ const EndpointListPage: React.FC<PropsWithChildren> = ({ children }) => {
       </Flex>
       <Table
         loading={{
-          spinning: isPendingPaginationAndFilter || isFilterPending,
+          spinning: isFilterPending || isPendingPageChange,
           indicator: <LoadingOutlined />,
         }}
         scroll={{ x: 'max-content' }}
         rowKey={'endpoint_id'}
-        dataSource={(sortedEndpointList || []) as Endpoint[]}
+        dataSource={(modelServiceList?.items || []) as Endpoint[]}
```

Comment:
Please use `filterNonNullItems` for `modelServiceList?.items`. and remove `as Endpoint[]`